include/envoy/router/router.h

```diff
@@ -232,6 +232,15 @@ class VirtualCluster {
 class RateLimitPolicy;
 class Config;
 
+/**
+ * All pre-processed per route config objects should be derived from this class.
+ */
+class PerRouteConfigObject {
```

Comment:
I think this can be named "PerRouteConfig", or maybe "RouteSpecificConfig". Or "RouteSpecificFilterConfig".

Reply:
I would vote for "RouteSpecificFilterConfig" but I don't feel that strongly about it.

Reply:
Now, this is going to be painful isn't it? Every filter will now have two config objects in some sense. a preprocessed per-route thing and a filter level config object. I guess thats okay, given that the perf benefit

Reply:
Every filter is basically doing this today, and it's entirely optional, so it seems OK to me?

Reply:
fair enough.